gt/main.c

```diff
@@ -153,15 +291,74 @@ gt_conf_put(struct gt_config *gt_conf)
 }
 
 static int
+config_gt_instance(struct gt_config *gt_conf, unsigned int lcore_id)
+{
+	int ret;
+	char lua_entry_path[128];
+	unsigned int block_idx = get_block_idx(gt_conf, lcore_id);
+	struct gt_instance *instance = &gt_conf->instances[block_idx];
+
+	ret = snprintf(lua_entry_path, sizeof(lua_entry_path), \
+			"%s/%s", LUA_POLICY_BASE_DIR, GRANTOR_CONFIG_FILE);
+	RTE_VERIFY(ret > 0 && ret < (int)sizeof(lua_entry_path));
+
+	instance->lua_state = luaL_newstate();
+	if (instance->lua_state == NULL) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"gt: failed to create new Lua state at lcore %u!\n",
+			lcore_id);
+		ret = -1;
+		goto out;
+	}
+
+	luaL_openlibs(instance->lua_state);
+	set_lua_path(instance->lua_state, LUA_POLICY_BASE_DIR);
+	ret = luaL_loadfile(instance->lua_state, lua_entry_path);
+	if (ret != 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"gt: %s!\n", lua_tostring(instance->lua_state, -1));
+		ret = -1;
+		goto out;
+	}
+
+	/* Call functions to initialize the policy table. */
+	ret = lua_pcall(instance->lua_state, 0, 0, 0);
+	if (ret != 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"gt: %s!\n", lua_tostring(instance->lua_state, -1));
+		ret = -1;
+		goto out;
+	}
+
+	/* Push functions. */
+	lua_getglobal(instance->lua_state, "setup_policy");
```

Comment:
Given that you already issued a `lua_pcall()` above, I think we can drop this second call, can't we? If a policy wants to initialize anything it would do in the previous call. What do you think?